src/tests/ftest/dfuse/bash_dcache.py

```diff
@@ -0,0 +1,82 @@
+"""
+  (C) Copyright 2024 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import os
+import stat
+
+from dfuse_test_base import DfuseTestBase
+from run_utils import run_remote
+
+SCRIPT = """#!/bin/bash
+
+#!/bin/bash
+
+dir_name="dir00"
+file_name="file00"
+
+rm -rf ${dir_name}
+mkdir ${dir_name}
+echo "Hello" > ${dir_name}/${file_name}
+rm -rf ${dir_name}
+
+mkdir ${dir_name}
+echo "Hello" > ${dir_name}/${file_name}
+cat ${dir_name}/${file_name}
+"""
+
+
+class DFuseBashdcacheTest(DfuseTestBase):
+    """Base Bashdcache test class.
+
+    :avocado: recursive
+    """
+
+    def run_bash_dcache_pil4dfs(self):
+        """Run a shell script which creates dir and file, then removes them and recreates.
+
+        This attempts to replicate the way that configure scripts repeating creating & removing
+        files under conftest.dir in bash.
+        """
+
+        lib_path = os.path.join(self.prefix, "lib64", "libpil4dfs.so")
+        env_str = f"export LD_PRELOAD={lib_path}; "
+
+        pool = self.get_pool(connect=False)
+        container = self.get_container(pool)
+        self.start_dfuse(self.hostlist_clients, pool, container)
+
+        fuse_root_dir = self.dfuse.mount_dir.value
+
+        with open(os.path.join(fuse_root_dir, "sh_dcache.sh"), "w") as fd:
+            fd.write(SCRIPT)
+
+        os.chmod(os.path.join(fuse_root_dir, "sh_dcache.sh"), stat.S_IXUSR | stat.S_IRUSR)
+
+        cmd = f"cd {fuse_root_dir}; ./sh_dcache.sh"
+
+        result = run_remote(self.log, self.hostlist_clients, env_str + cmd)
+        if not result.passed:
+            self.fail(f'"{cmd}" failed on {result.failed_hosts}')
+
+        # Turn on dcache in bash
+        env_str = env_str + "export D_IL_NO_DCACHE_BASH=0; "
+
+        result = run_remote(self.log, self.hostlist_clients, env_str + cmd)
+        if result.passed:
+            self.fail(f'"{cmd}" failed on {result.failed_hosts}')
```

Comment:
How does this fail?

Reply:
"cat" fails 
cat: dir00/file00: No such file or directory

Reply:
is that from the file or the directory do you know?

Does this test run with IL logging enabled and if so can you share the link please?

Reply:
The directory does exist since we call "mkdir" to re-create it. The cached object for the directory in bash process is probably wrong.
The error occurs in open() inside "cat". The new process does not out-of-date cached dir info. It failed because the file does not exist. The logging was not enabled.

Reply:
So is the new file written but goes to the deleted/non-existent directory?

Reply:
Right.